src/main/java/jenkins/plugins/slack/StandardSlackService.java

```diff
@@ -88,6 +101,26 @@ public boolean publish(String message, String color) {
         return result;
     }
 
+    private String getTokenToUse() {
+        if (authTokenCredentialId != null && !authTokenCredentialId.isEmpty()) {
+            StringCredentials credentials = lookupCredentials(authTokenCredentialId);
+            if (credentials != null) {
+                logger.info("Using Integration Token Credential ID.");
+                return credentials.getSecret().getPlainText();
+            }
+        }
+
+        logger.info("Using Integration Token.");
+
+        return token;
+    }
+
+    private StringCredentials lookupCredentials(String credentialId) {
+        List<StringCredentials> credentials = CredentialsProvider.lookupCredentials(StringCredentials.class, Jenkins.getInstance(), ACL.SYSTEM, Collections.<DomainRequirement>emptyList());
```

Comment:
Obviously this works, but what about something like this (after all, the person configuring the plugin should be aware of the correct credential id to use):
```java
IdCredentials cred = CredentialsProvider.findCredentialById(credentialsId, IdCredentials.class, build);
        if (cred==null)
            throw new CredentialNotFoundException(credentialsId);

        if (StringCredentials.class.isInstance(cred))
            return StringCredentials.class.cast(cred);
```

For example, see: https://github.com/jenkinsci/credentials-binding-plugin/blob/master/src/main/java/org/jenkinsci/plugins/credentialsbinding/MultiBinding.java#L122